### PR TITLE
Security

### PR DESCRIPTION
**Pull request by ToStefan**

### CODE REVIEW

File: src/main/java/hybrid/it/internship/library/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@EnableGlobalMethodSecurity(
+        prePostEnabled = true,
+        securedEnabled = true,
+        jsr250Enabled = true)
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/config/SecurityConfig.java]: Use constructor wiring instead

**Review by mspavkovic — APPROVED** [review]

File: src/main/java/hybrid/it/internship/library/security/AuthenticationEntryPoint.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    public void commence(final HttpServletRequest request,
+                         final HttpServletResponse response,
+                         final AuthenticationException authException) throws IOException {
+
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        response.addHeader("WWW-Authenticate", "Basic realm=" + getRealmName() + "");
+
+        PrintWriter writer = response.getWriter();
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/security/AuthenticationEntryPoint.java]: Use slf4J instead..

File: src/main/java/hybrid/it/internship/library/security/UserPrincipal.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                .collect(Collectors.toList());
+
+        return new UserPrincipal(
+                user.getUsername(),
+                user.getPassword(),
+                authorities
+        );
+    }
+
+    public UserPrincipal(String username, String password, List<GrantedAuthority> roles) {
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/security/UserPrincipal.java]: put constructor on top

File: src/main/java/hybrid/it/internship/library/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -21,6 +24,11 @@
     private final UserRepository userRepository;
     private final UserMapper userMapper;
 
+    @Bean
```
Comment (nemanja-blagojevic-valcon) [inline src/main/java/hybrid/it/internship/library/service/impl/UserServiceImpl.java]: Use classical wiring..

File: src/main/java/hybrid/it/internship/library/security/UserPrincipal.java [hunk 5]
```diff
@@ -11,6 +13,8 @@
 
 public class UserPrincipal implements UserDetails {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/security/UserPrincipal.java]: use @Slf4j provided by lombok

File: src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java [hunk 6]
```diff
@@ -22,6 +24,8 @@
 @Service
 public class RentServiceImpl implements RentService {
```
Comment (babicmilan) [inline src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java:26]: same

**Review by babicmilan — CHANGES_REQUESTED** [review]

File: src/main/java/hybrid/it/internship/library/LibraryApplication.java [hunk 7]
```diff
@@ -1,11 +1,15 @@
 package hybrid.it.internship.library;
 
+import org.slf4j.Logger;
```
Comment (mspavkovic) [inline src/main/java/hybrid/it/internship/library/LibraryApplication.java]: remove not used imports

File: src/main/java/hybrid/it/internship/library/security/UserPrincipal.java [hunk 8]
```diff
@@ -11,6 +13,8 @@
 
 public class UserPrincipal implements UserDetails {
```
Comment (mspavkovic) [inline src/main/java/hybrid/it/internship/library/security/UserPrincipal.java:12]: Use @Slf4j lombok annotation for logging https://projectlombok.org/features/log

File: src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java [hunk 9]
```diff
@@ -22,6 +24,8 @@
...
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (mspavkovic) [inline src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java]: same as above

**Review by nemanja-blagojevic-valcon — APPROVED** [review]

**Review by mspavkovic — APPROVED** [review]

**Pull request merged**